nixos/modules/services/networking/frp-server.nix

```diff
@@ -0,0 +1,84 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.frp-server;
+  settingsFormat = pkgs.formats.ini { };
+in
+{
+  options = {
+    services.frp-server = {
+      enable = mkEnableOption (lib.mdDoc "frp server configuration");
+
+      package = mkPackageOptionMD pkgs "frp" { };
+
+      settings = mkOption {
+        type = settingsFormat.type;
+        default = { };
+        description = lib.mdDoc ''
+          frp server configuration, for configuration options see the example on [github](https://github.com/fatedier/frp/blob/dev/conf/frps_full.ini)
+          note that you have to do the firewall configuration manually
+        '';
+        example = literalExpression ''
+          {
+            common = {
+              bind_port = 7000;
+            };
+          }
+        '';
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    environment = {
+      etc."frps.ini".source = settingsFormat.generate "frps.ini" cfg.settings;
+    };
+    systemd.services = {
+      frp-server = {
+        wants = [ "network.target" ];
```

Comment:
```suggestion
```

not needed, see `man systemd.special`

Reply:
Are you referring to this section?
```
Note the distinction between this unit and network.target. This unit is an active unit (i.e. pulled in by the consumer rather than the provider of this functionality) and pulls in a service which
           possibly adds substantial delays to further execution. In contrast, network.target is a passive unit (i.e. pulled in by the provider of the functionality, rather than the consumer) that usually
           does not delay execution much. Usually, network.target is part of the boot of most systems, while network-online.target is not, except when at least one unit requires it. Also see Running
           Services After the Network Is Up[1] for more information.
```